### DISCUSSION TITLE
Support for new virtual desktop custom wallpaper feature

### DISCUSSION THREAD

**Discussion by 0mri** [t1m0thyj/WinDynamicDesktop#340] [Q&A] [answered]
Hello guys,
there is a new feature on [Windows Insider Preview](https://blogs.windows.com/windows-insider/2021/03/17/announcing-windows-10-insider-preview-build-21337/) that let you set a custom wallpaper for each virtual desktop.

by default WinDD changes the wallpaper on the current active virtual desktop, others desktops stay with same background as before.

i write a simple ps script that change the wallpaper for each virtual desktop

```
$current_wallpaper = (Get-ItemProperty -Path 'Registry::HKEY_CURRENT_USER\Control Panel\Desktop'  -Name Wallpaper | Select-Object Wallpaper).Wallpaper

Get-ChildItem -Path Registry::HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Explorer\VirtualDesktops\Desktops | Set-ItemProperty -Name Wallpaper -Value $current_wallpaper
```
the problem is that the background isn't refreshes until i'm  restart Windows Explorer 

there is another way i can set new wallpaper for all desktops?

like this function:
![image](https://user-images.githubusercontent.com/29231457/121960252-836d3100-cd6e-11eb-9fdc-1f78230a0e56.png)

**Comment by t1m0thyj** [comment, marked as answer]
Changing the Wallpaper property in the Windows registry won't update it immediately as you've discovered.

Either Microsoft needs to publish an API to change the wallpaper across all desktops, or someone needs to reverse engineer the undisclosed API that already exists.

The latter seems likely to happen in https://github.com/Grabacr07/VirtualDesktop/pull/57, but I haven't been able to get it to work yet.

You can follow that PR or [this issue](https://github.com/t1m0thyj/WinDynamicDesktop/issues/321) to be informed when the situation changes.
